File: src/adapters/repo_manager.py
"""
Repo Manager - Gerencia operações Git e GitHub.
"""
from pathlib import Path
from typing import Optional
from libs.git_operations import GitOperations
import httpx

class RepoManager:
    """Gerenciador de repositórios Git."""
    
    def __init__(self, config):
        self.config = config
        self.git_ops = GitOperations()
    
    def clone_repo(self, repo_url: str, branch: Optional[str] = None) -> Path:
        """Clona repositório."""
        target = self.config.workdir / "repo"
        return self.git_ops.clone(repo_url, target, branch).working_dir
    
    def init_repo(self, path: Path):
        """Inicializa novo repositório."""
        self.git_ops.init(path)
    
    def create_branch(self, repo_path: Path, branch_name: str):
        """Cria nova branch."""
        import git
        repo = git.Repo(repo_path)
        self.git_ops.create_branch(repo, branch_name)
    
    def commit(self, repo_path: Path, message: str):
        """Faz commit."""
        import git
        repo = git.Repo(repo_path)
        self.git_ops.commit(repo, message)
    
    def create_pull_request(
        self,
        repo_url: str,
        title: str,
        body: str,
        branch: str
    ) -> Optional[str]:
        """Cria PR no GitHub."""
        # Extrair owner/repo da URL
        parts = repo_url.rstrip("/").split("/")
        owner, repo = parts[-2], parts[-1].replace(".git", "")
        
        # GitHub API
        headers = {
            "Authorization": f"token {self.config.github_token}",
            "Accept": "application/vnd.github+json"
        }
        
        data = {
            "title": title,
            "body": body,
            "head": branch,
            "base": self.config.default_base_branch,
            "draft": self.config.create_draft_pr
        }
        
        try:
            response = httpx.post(
                f"https://api.github.com/repos/{owner}/{repo}/pulls",
                headers=headers,
                json=data
            )
            if response.status_code == 201:
                return response.json()["html_url"]
        except Exception as e:
            print(f"Erro ao criar PR: {e}")
        
        return None

File: src/adapters/repoManager.ts
import * as path from 'path';
import * as fs from 'fs';
import simpleGit, { SimpleGit } from 'simple-git';
import axios from 'axios';
import { AppConfig } from '../types';

interface PullRequestOptions {
  repoUrl: string;
  title: string;
  body: string;
  branch: string;
}

export class RepoManager {
  constructor(private config: AppConfig) {}

  async cloneRepo(repoUrl: string, branch?: string): Promise<string> {
    const targetPath = path.join(this.config.workdir, 'repo');
    fs.mkdirSync(targetPath, { recursive: true });
    const git = simpleGit();
    const cloneOptions = branch ? ['--branch', branch] : [];
    await git.clone(repoUrl, targetPath, cloneOptions);
    return targetPath;
  }

  async initRepo(repoPath: string): Promise<void> {
    const git = simpleGit(repoPath);
    await git.init();
    await git.addConfig('user.name', this.config.defaultAuthor);
    await git.addConfig('user.email', 'enginai@local');
  }

  async createBranch(repoPath: string, branchName: string): Promise<void> {
    const git: SimpleGit = simpleGit(repoPath);
    await git.checkoutLocalBranch(branchName);
  }

  async commit(repoPath: string, message: string): Promise<void> {
    const git: SimpleGit = simpleGit(repoPath);
    await git.add('.');
    await git.commit(message, ['--allow-empty']);
  }

  async pushBranch(repoPath: string, repoUrl: string, branchName: string): Promise<void> {
    const git: SimpleGit = simpleGit(repoPath);
    // Inject token into remote URL for authenticated push
    const authedUrl = repoUrl.replace(
      'https://',
      `https://${this.config.githubToken}@`,
    );
    await git.remote(['set-url', 'origin', authedUrl]);
    await git.push('origin', branchName, ['--set-upstream']);
  }

  async createPullRequest(opts: PullRequestOptions): Promise<string | null> {
    const parts = opts.repoUrl.replace(/\.git$/, '').split('/');
    const owner = parts[parts.length - 2];
    const repo = parts[parts.length - 1];

    try {
      const response = await axios.post(
        `https://api.github.com/repos/${owner}/${repo}/pulls`,
        {
          title: opts.title,
          body: opts.body,
          head: opts.branch,
          base: this.config.defaultBaseBranch,
          draft: this.config.createDraftPr,
        },
        {
          headers: {
            Authorization: `token ${this.config.githubToken}`,
            Accept: 'application/vnd.github+json',
          },
        },
      );
      return response.data.html_url as string;
    } catch (err) {
      console.error(`Failed to create PR: ${(err as Error).message}`);
      return null;
    }
  }
}
